src/lib/radio/_radio-theme.scss

```diff
@@ -12,28 +12,64 @@
   .mat-radio-outer-circle {
     border-color: mat-color($foreground, secondary-text);
 
-    .mat-radio-checked & {
-      border-color: mat-color($accent);
-    }
-
     .mat-radio-disabled & {
       border-color: mat-color($foreground, disabled);
     }
   }
 
   .mat-radio-inner-circle {
-    background-color: mat-color($accent);
-
     .mat-radio-disabled & {
       background-color: mat-color($foreground, disabled);
     }
   }
 
   .mat-radio-ripple .mat-ripple-element {
-    background-color: mat-color($accent, 0.26);
-
     .mat-radio-disabled & {
       background-color: mat-color($foreground, disabled);
     }
   }
+
+  .mat-radio-button.mat-radio-checked {
+    &.mat-primary {
+      .mat-radio-outer-circle{
+        border-color: mat-color($primary);
+      }
+
+      .mat-radio-inner-circle {
+        background-color: mat-color($primary);
+      }
+
+      .mat-radio-ripple .mat-ripple-element {
+        background-color: mat-color($primary, 0.26);
+      }
+    } 
+
+    &.mat-accent {
+      .mat-radio-outer-circle{
+        border-color: mat-color($accent);
+      }
+
+      .mat-radio-inner-circle {
+        background-color: mat-color($accent);
+      }
+
+      .mat-radio-ripple .mat-ripple-element {
+        background-color: mat-color($accent, 0.26);
+      }
+    } 
+
+    &.mat-warn {
+      .mat-radio-outer-circle{
+        border-color: mat-color($warn);
+      }
+
+      .mat-radio-inner-circle {
+        background-color: mat-color($warn);
+      }
+
+      .mat-radio-ripple .mat-ripple-element {
+        background-color: mat-color($warn, 0.26);
+      }
+    } 
```

Comment:
I have the feeling that we could move that into a mixin? So we don't have to repeat code three times.